pyccl/cls.py

```diff
@@ -316,18 +328,18 @@ def angular_cl(cosmo, cltracer1, cltracer2, ell,
     if isinstance(ell, float) or isinstance(ell, int):
         # Use single-value function
         cl_one, status = lib.angular_cl_vec(
-            cosmo, clt1, clt2, l_limber, l_logstep,
+            cosmo, clt1, clt2, psp, l_limber, l_logstep,
             l_linstep, [ell], 1, status)
         cl = cl_one[0]
     elif isinstance(ell, np.ndarray):
         # Use vectorised function
         cl, status = lib.angular_cl_vec(
-            cosmo, clt1, clt2, l_limber, l_logstep,
+            cosmo, clt1, clt2, psp, l_limber, l_logstep,
             l_linstep, ell, ell.size, status)
     else:
```

Comment:
While you are here, using `np.atleast_1d` will cover all three of these cases in this if else block

Reply:
Done in last commit. Note that there's a lot of these in most of the python functions.